utils/S3UtilsMongoClient.js

```diff
@@ -0,0 +1,246 @@
+const { MongoClientInterface } = require('arsenal').storage.metadata.mongoclient;
+const { validStorageMetricLevels } = require('../CountItems/utils/constants');
+
+class S3UtilsMongoClient extends MongoClientInterface {
+    getObjectMDStats(bucketName, bucketInfo, isTransient, log, callback) {
+        const c = this.getCollection(bucketName);
+        const cursor = c.find({}, {
+            projection: {
+                '_id': 1,
+                'value.last-modified': 1,
+                'value.replicationInfo': 1,
+                'value.dataStoreName': 1,
+                'value.content-length': 1,
+                'value.versionId': 1,
+                'value.owner-display-name': 1,
+                'value.isDeleteMarker': 1,
+                'value.isNull': 1,
+            },
+        });
+        const collRes = {
+            bucket: {}, // bucket level metrics
+            location: {}, // location level metrics
+            account: {}, // account level metrics
+        };
+        let stalledCount = 0;
+        const cmpDate = new Date();
+        cmpDate.setHours(cmpDate.getHours() - 1);
+
+        cursor.forEach(
+            res => {
+                const { data, error } = this._processEntryData(bucketName, res, isTransient);
+
+                if (error) {
+                    log.error('Failed to process entry data', {
+                        method: 'getObjectMDStats',
+                        entry: res,
+                        error,
+                    });
+                }
+
+                let targetCount;
+                let targetData;
+                if (res._id.indexOf('\0') !== -1) {
+                    // versioned item
+                    targetCount = 'versionCount';
+                    targetData = 'versionData';
+
+                    if (res.value.replicationInfo.backends.length > 0
+                        && this._isReplicationEntryStalled(res, cmpDate)) {
+                        stalledCount++;
+                    }
+                } else if (!!res.value.versionId && !res.value.isNull) {
```

Comment:
Why do we check for isNull here?

From what I remember, "null versions" reference the objects located in an unversioned bucket.

Reply:
the previous `countItmes` service has an issue when bucket is suspended, because of [this logic ](https://github.com/scality/Arsenal/blob/development/8.1/lib/storage/metadata/mongoclient/MongoClientInterface.js#L2077) when location's nonCurrent size is got by versioned size minus master size(it works in versioned bucket, because master data are duplicated), but the object uploaded after the bucket is suspended will only have master data in DB, so this record is not duplicated so should not be subtracted

that's why I take out of `isNull` data and treat it as non versioned data

Reply:
Would you mind adding a comment here?

Reply:
yeah, no problem
Thanks for reviewing!